src/vip/batch_address_test_tool/processor.clj

```diff
@@ -0,0 +1,135 @@
+(ns vip.batch-address-test-tool.processor
+  (:require [clojure.tools.logging :as log]
+            [clojure.data.csv :as csv]
+            [vip.batch-address-test-tool.queue :as q]
+            [vip.batch-address-test-tool.cloud-store :as cloud-store]
+            [clojure.string :as str])
+  (:import [java.io File]))
+
+(defn ->pass-through
+  "Defines a function that will only call source-fn if there isn't an error
+  in the context"
+  [source-fn]
+  (fn [ctx]
+    (if (contains? ctx :error)
+      ctx
+      (source-fn ctx))))
+
+(defn retrieve-file
+  "Retrieves the file for processing from s3"
+  [ctx]
+  (try
+    (let [file-name   (get-in ctx [:input "fileName"])
+          bucket-name (get-in ctx [:input "bucketName"])
+          file        (cloud-store/fetch-file bucket-name file-name)]
+      (assoc ctx :address-file file))
+    (catch Exception ex
+      (assoc ctx :error ex))))
+
+(defn validate-header-row
+  "Validates the header row of the input file to confirm 2 headers with
+   expected names"
+  [header-row]
+  (if (not= (count header-row) 2)
+    (throw (Exception. "Incorrect number of headers, expected 2"))
+    (if (not= (first header-row) "voter_address")
+      (throw (Exception. "Expected first header item to be \"voter_address\""))
+      (if (not= (second header-row) "expected_polling_location")
+        (throw (Exception. "Expected second header item to be \"expected_polling_location\""))))))
+
+(defn validate-and-parse-row
+  "Validates number of elements in a row and returns row values parsed into
+   a map"
+  [row]
+  (if (not= (count row) 2)
+    (throw (Exception. "Incorrect number of row elements, expected 2"))
+    {:address (first row)
+     :expected-polling-location (second row)}))
+
+(defn validate-and-parse-file*
+  "Calls validate-header-row on first row in file and maps remaining rows with
+   validate-and-parse-row"
+  [ctx]
+  (try
+    (let [rows (csv/read-csv (:address-file ctx))]
+      (validate-header-row (first rows))
+      ; TODO iterate with index to stop at 300 rows
```

Comment:
What about something like `(take 301 (csv/read-csv ,,,)` instead of keeping an index?